### ISSUE TITLE
New eventlet release 0.34.1

### ISSUE THREAD

**Issue by 4383** [eventlet/eventlet#842]
As significant steps have been done I'd suggest releasing a new version of eventlet, 0.34.0.

- #822
- #817 
- #820 
- #827 
- #831 
- #832 
- #841

Why a minor version and not a bug fix version or a major version?

Because support of Python 2.7 have been removed and support of Python 3.12 have been added.

We can consider that backward compatibility is now broken, due to the removing of the support of Python 2.7, usually it would require a major release, but as eventlet is still not considered as stable, higher than version 1.0.0, I won't recommend a major version here.

Also a bug fix version won't reflect the support removal.

So a minor version remain our better option.

Are you ok with that?

If yes, does someone volunteer to create the related tag and run the mechanisms to upload to pypi?

**Comment by itamarst** [comment]
Probably doing https://blog.pypi.org/posts/2023-04-20-introducing-trusted-publishers/ is better than manual upload, especially if long term goal is to focus on stable maintenance only, so the less work the better.

**Comment by itamarst** [comment]
I can try to set that infrastructure up next week, I've done it a couple of times.

**Comment by 4383** [comment]
> Probably doing https://blog.pypi.org/posts/2023-04-20-introducing-trusted-publishers/ is better than manual upload, especially if long term goal is to focus on stable maintenance only, so the less work the better.

Yeah, that's I was thinking (OIDC), that's what I made on all my recent github project, including on aiohub:

https://github.com/4383/aiohub/blob/main/.github/workflows/main.yml#L52

We don't want to publish manually, but we want a bit of coordination for tag creation, that's the goal of my proposal here https://github.com/eventlet/eventlet/issues/843

**Comment by 4383** [comment]
> I can try to set that infrastructure up next week, I've done it a couple of times.

As you want, I can help you if you want. Let me know :)

**Comment by 4383** [comment]
I configured Pypi to use OIDC and I just proposed a patch to publish from github actions on tag creation

**Comment by itamarst** [comment]
Also blocked on #846.

**Comment by itamarst** [comment]
Having read the discussion in #754, at least part of that needs to make it in before release, specifically the change from `==` to `is`.

**Comment by 4383** [comment]
We also to do another round of updates against the changelog (NEWS) to reflect all the recent changes.

**Comment by 4383** [comment]
@itamarst: I think we are ready to go, please can you double check and confirm?

**Comment by 4383** [comment]
@itamarst: Waiting for your explicit confirmation before pushing the new tag. Just making my waiting explicit. I don't want to harass you, no rush :)

**Comment by itamarst** [comment]
Go for it, thank you!

**Comment by 4383** [comment]
@jstasiak: o/ Sorry to disturb you again with credential stuffs, but unfortunately I'm still not authorized to push directly to the eventlet repo, so I can't push new tags. For the current workflow (pull requests) that's not an issue, I always works from my fork, but for infra and maintenance actions that's an issue. 

Please can you give me the push power to the repo?

I think @itamarst's github profile need the same action.

**Comment by 4383** [comment]
Cc @temoto

**Comment by 4383** [comment]
Fixed. My bad :) 

My ssh key was ignored and I wondered why, I forgot to update my remote from `https://github.com/eventlet/eventlet` to `git@github.com:eventlet/eventlet`... sorry

**Comment by 4383** [comment]
Now the tag is there https://github.com/eventlet/eventlet/releases/tag/v0.34.0

**Comment by 4383** [comment]
The publish action has been ignored, I think its due to the wrong naming of the action file, should fix it #852.
I'll delete the recently pushed tag and recreate a new one once this fix will be merged

**Comment by 4383** [comment]
@temoto I doesn't have the permissions to delete https://github.com/eventlet/eventlet/releases/tag/v0.34.0

**Comment by jstasiak** [comment]
I just deleted the tag. We'll have to figure out the permissions

**Comment by itamarst** [comment]
It's also easy to just have 0.34.1, .2, etc. until GHA is debugged (from personal experience).

**Comment by jstasiak** [comment]
I think this was a one-time kind of issue anyway (although it'd be great if one could trigger the release from the GH Actions interface? Seems bad to rely on it always working on the first try)

**Comment by 4383** [comment]
I fixed the GHA => https://github.com/eventlet/eventlet/pull/852 Now the deployment job is well triggered. That was a first real try I don't think we will see new similar errors.

However, if you prefer I can publish a 0.34.1, but people could ask where is 0.34.0, just want to avoid that.

**Comment by 4383** [comment]
@temoto: After careful consideration, if you don't mind, I prefer you remove the tag `0.34.0`. I'll recreate it and hence rerun the pipeline. We can give another try to this approach, and, if we face a new issue again, then, in this case, I'll iterate over new releases as suggested by Itamar.

We must avoid removing artifacts and versions from Pypi, and new iterate over tags, would require changelog updates each time to explain the bit and the bytes of these new debugging versions. Not sure it would be more easy...

I'm confident concerning my fix #852, and the new publishing should works fine. Hopefully I'm right :)

**Comment by 4383** [comment]
I see you already removed the tag, thanks.

**Comment by 4383** [comment]
Almost there... https://github.com/eventlet/eventlet/pull/855

Unfortunatelly I was wrong, it failed again :)
I seen that the rerun button appeared on my dashboard, however, if I rerun the failing job (https://github.com/eventlet/eventlet/actions/runs/7278151715), I'm not sure if actions will be pulled from master or from the tag's sources. If actions are pulled from the tag, then, the error will thrown again, because the fixed version will be on master and would require a new tag to be considered.

Any ideas?

Anyway, as I said previously, I'm ok to produce a `v0.34.1`.

**Comment by 4383** [comment]
Changelog updated to reflect the changes landed in `v0.34.1` https://github.com/eventlet/eventlet/pull/856

`v0.34.0` won't be available on Pypi as I'm not able to remove the existing tag from github to replace it with a new tag that correspond to the new changes merged recently. These recent changes will be in `v0.34.1`.

**Comment by 4383** [comment]
Done, new pypi release successfully created https://pypi.org/project/eventlet/0.34.1/

Thanks everyone for your help

**Comment by 1D0BE** [comment]
I think there is no eventlet.__version__ now. Just tried to start a gunicorn server with it...

**Comment by 4383** [comment]
Yes `eventlet.__version__` has been removed:
https://github.com/eventlet/eventlet/commit/33e05cae342418333dd7ed649d21b77621895585

**Comment by 4383** [comment]
`pip show eventlet` should be prefered, or even your requirements file.

**Comment by 1D0BE** [comment]
ok makes sense, thanks

**Comment by 4383** [comment]
Else, we could bring back something like:

```
from importlib.metadata import version as getversion

__version__ = getversion("eventlet")
```

But I'm not convinced that's really needed.

**Comment by 4383** [comment]
@1D0BE: Fix proposed https://github.com/eventlet/eventlet/pull/859
